examples/bri-3/src/bri/identity/bpiSubjects/models/bpiSubject.ts

```diff
@@ -1,8 +1,9 @@
 import { AutoMap } from '@automapper/classes';
 import { v4 } from 'uuid';
+import { IDomainObject } from '../../../../shared/domainObject.interface.js';
 import { BpiSubjectRole } from './bpiSubjectRole';
 
-export class BpiSubject {
+export class BpiSubject implements IDomainObject  {
```

Comment:
why is it necessary to have the implemented interface IDomainObject and private activator<T extends IDomainObject>. Just to ensure a common inherited value of id?

Reply:
The interface is not needed at all. I was following an article and thought that it might be a good idea in order to make sure only domain objects are accepted for activation but on second thought i think it is over engineering. Removed.
The activator is necessary to avoid instantiation of each domain object ctor by providing empty params. I was hoping for a fully generic solution (one method for any mapping pair) but couldn't make it work so stopped here for now.